provider/kubernetes/kubernetes.go

```diff
@@ -190,161 +192,214 @@ func (p *Provider) loadIngresses(k8sClient Client) (*types.Configuration, error)
 		}
 		templateObjects.TLS = append(templateObjects.TLS, tlsSection...)
 
-		for _, r := range i.Spec.Rules {
-			if r.HTTP == nil {
-				log.Warn("Error in ingress: HTTP is nil")
+		templateObjects, err = p.buildTemplateFromIngress(i, templateObjects, k8sClient)
+		if err != nil {
+			log.Errorf("Cannot build Ingress %s/%s due to error: %v", i.Namespace, i.Name, err)
+			continue
+		}
+
+		err = p.updateIngressStatus(i, k8sClient)
+		if err != nil {
+			log.Errorf("Cannot update Ingress %s/%s due to error: %v", i.Namespace, i.Name, err)
+		}
+	}
+	return &templateObjects, nil
+}
+
+func (p *Provider) buildTemplateFromIngress(i *extensionsv1beta1.Ingress, templateObjects types.Configuration, k8sClient Client) (types.Configuration, error) {
+	if i.Spec.Backend != nil {
+		// If a global backend is configured, add this as a "default route" by adding a wildcard frontend, with the priority of the ingress.
+
+		templateObjects, err := createEmptyBackend(defaultBackendName, templateObjects)
+		if err != nil {
+			return templateObjects, fmt.Errorf("cannot create empty backend: %s", err)
+		}
+
+		service, exists, err := k8sClient.GetService(i.Namespace, i.Spec.Backend.ServiceName)
+		if err != nil {
+			return templateObjects, fmt.Errorf("error while retrieving service information from k8s API %s/%s: %v", i.Namespace, i.Spec.Backend.ServiceName, err)
+		}
+		if !exists {
+			return templateObjects, fmt.Errorf("service not found for %s/%s", i.Namespace, i.Spec.Backend.ServiceName)
+		}
+
+		templateObjects, err = setBackendSettingsFromServiceAnnotations(defaultBackendName, service, templateObjects)
+		if err != nil {
+			return templateObjects, fmt.Errorf("cannot create settings for backend: %s", err)
+		}
+
+		endpoints, exists, err := k8sClient.GetEndpoints(service.Namespace, service.Name)
+		if err != nil {
+			return templateObjects, fmt.Errorf("error retrieving endpoint information from k8s API %s/%s: %v", service.Namespace, service.Name, err)
+		}
+		if !exists {
+			log.Warnf("Endpoints not found for %s/%s", service.Namespace, service.Name)
+		}
+		if len(endpoints.Subsets) == 0 {
+			log.Warnf("Endpoints not available for %s/%s", service.Namespace, service.Name)
+		}
+
+		for _, subset := range endpoints.Subsets {
+			endpointPort := endpointPortNumber(corev1.ServicePort{Protocol: "TCP", Port: int32(i.Spec.Backend.ServicePort.IntValue())}, subset.Ports)
```

Comment:
It seems to be still valid to specify the service port as string, referencing the actual service port. Therefore, I think we have to write a second function that does that logic for the ingress service port. WDYT?

Reply:
You are correct, but its even simpler for this spec! 😄

Reply:
Tests added!